drivers/pwm/pwm_stm32.c

```diff
@@ -140,20 +119,16 @@ static int pwm_stm32_get_cycles_per_sec(struct device *dev, u32_t pwm,
 {
 	const struct pwm_stm32_config *cfg = DEV_CFG(dev);
 	struct pwm_stm32_data *data = DEV_DATA(dev);
-	u32_t bus_clk, tim_clk;
+	u32_t bus_clk;
 
 	if (cycles == NULL) {
 		return -EINVAL;
 	}
 
-	/* Timer clock depends on APB prescaler */
 	clock_control_get_rate(data->clock,
 			(clock_control_subsys_t *)&cfg->pclken, &bus_clk);
 
-	tim_clk = __get_tim_clk(bus_clk,
-			(clock_control_subsys_t *)&cfg->pclken);
-
-	*cycles = (u64_t)(tim_clk / (data->pwm_prescaler + 1));
+	*cycles = (u64_t)(bus_clk / (data->pwm_prescaler + 1));
```

Comment:
No, this is wrong. But on the other side, there is also a bug in __get_tim_clk(), so...
If you look in the "Timer clock" section under RCC in the datasheet (6.2.15 in my case for a STM32L4x6), the timer clock is
1. equal to the bus clock if APB prescaler is 1
2. twice the bus clock otherwise

Thus you miss the second case. Try with an oscilloscope and change PPRE1 or PPRE2 depending on your timer.